### PR TITLE
[#163019561] Admin to filter security users by status

### PR DESCRIPTION
**Pull request by nelmandela**
### What does PR do?

	•Allows admin to filter security users by status

### Description of task to be completed:

	•Add `active` field to the SecurityUser model
        •Add security_user filters
        •Add test

### How can it be tested manually?

	•Navigate to the `security_user` endpoint & test out the filters

### Related pivotal tracker stories

	[#163019561](https://www.pivotaltracker.com/story/show/163019561) 

### Screenshots:

screen shot 2019-01-31 at 12 02 17

### CODE REVIEW

File: core/models/user.py [hunk 1]
```diff
@@ -89,6 +89,7 @@ def _create_assignee_object_for_user(self):
 
 class SecurityUser(User):
     badge_number = models.CharField(max_length=30, unique=True)
+    active = models.BooleanField(default=True)
```
Comment (egichuri) [inline core/models/user.py]: the User model comes with an `is_active` field. we can use that for consistency.

File: api/tests/test_security_user.py [hunk 2]
```diff
@@ -127,3 +127,14 @@ def test_security_user_api_endpoint_cant_allow_delete(self, mock_verify_token):
             HTTP_AUTHORIZATION="Token {}".format(self.token_admin),
         )
         self.assertEqual(response.data, {'detail': 'Method "DELETE" not allowed.'})
+
+    @patch('api.authentication.auth.verify_id_token')
+    def test_admin_can_filter_security_users_by_status(self, mock_verify_id_token):
+        mock_verify_id_token.return_value = {'email': self.admin_user.email}
+        # url = reverse('security_users-list')
```
Comment (egichuri) [inline api/tests/test_security_user.py]: remove commented out stuff :)

File: api/tests/test_security_user.py [hunk 3]
```diff
@@ -127,3 +127,14 @@ def test_security_user_api_endpoint_cant_allow_delete(self, mock_verify_token):
...
+        response = client.get(
+            '{}?active={}'.format(self.security_users_admin_url, self.security_user.active),
+            HTTP_AUTHORIZATION="Token {}".format(self.token_admin),
```
Comment (egichuri) [inline api/tests/test_security_user.py]: For this test let's confirm `self.security_user` is in the response. also  confirm the user isn't in the response for a filter with `not self.security_user.active`

Comment (nelmandela) [reply api/tests/test_security_user.py]: Alright

**Comment by nelmandela** [comment]
How does this command work "isort -rc --check-only --diff --atomic -q"
i'm running it but nothing is happening

**Comment by nelmandela** [comment]
@egichuri done

File: api/tests/test_security_user.py [hunk 4]
```diff
@@ -127,3 +127,28 @@ def test_security_user_api_endpoint_cant_allow_delete(self, mock_verify_token):
...
+    def test_user_not_in_filter_of_in_active_security_user(self, mock_verify_id_token):
+        mock_verify_id_token.return_value = {'email': self.admin_user.email}
+        response = client.get(
+            '{}?active={}'.format(
+                self.security_users_admin_url, not self.security_user.is_active
+            ),
+            HTTP_AUTHORIZATION="Token {}".format(self.token_admin),
+        )
+        self.assertEqual(response.status_code, 200)
+        self.assertNotEqual(response.data, self.security_user.email)
```
Comment (egichuri) [inline api/tests/test_security_user.py]: and `assertNotIn` here

File: api/tests/test_security_user.py [hunk 5]
```diff
@@ -127,3 +127,28 @@ def test_security_user_api_endpoint_cant_allow_delete(self, mock_verify_token):
...
+        mock_verify_id_token.return_value = {'email': self.admin_user.email}
+        response = client.get(
+            '{}?active={}'.format(
+                self.security_users_admin_url, self.security_user.is_active
+            ),
+            HTTP_AUTHORIZATION="Token {}".format(self.token_admin),
+        )
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(response.data['count'], 1)
+        self.assertEqual(response.data['results'][0]['email'], self.security_user.email)
```
Comment (egichuri) [inline api/tests/test_security_user.py]: tiny last request - let's use `assertIn` to avoid this failing if more security users are added in setup

Comment (nelmandela) [reply api/tests/test_security_user.py]: okay

**Comment by egichuri** [comment]
@nelmandela you've missed this https://github.com/AndelaOSP/art-backend/pull/245/files#diff-7a729f127c71c1d96d0faecf03df317dR154

also delete the 2 migration files

**Comment by egichuri** [comment]
**also delete the 2 migration files** - one was for adding a field and the second one for removing it, so we need not commit it

**Pull request merged**